lib/supraffi/seal.go

```diff
@@ -137,6 +163,61 @@ func SupraSealInit(sectorSize uint64, configFile string) {
 	C.supra_seal_init(C.size_t(sectorSize), cConfigFile)
 }
 
+// GetHealthInfo retrieves health information for all NVMe devices
+func GetHealthInfo() ([]HealthInfo, error) {
+	// Allocate space for raw C struct
+	const maxControllers = 64
+	rawInfos := make([]C.nvme_health_info_t, maxControllers)
+
+	// Get health info from C
+	count := C.get_nvme_health_info(
+		(*C.nvme_health_info_t)(unsafe.Pointer(&rawInfos[0])),
+		C.size_t(maxControllers),
+	)
+
+	if count == 0 {
+		return nil, fmt.Errorf("no NVMe controllers found")
+	}
+
+	// Convert C structs to Go structs
+	healthInfos := make([]HealthInfo, count)
+	for i := 0; i < int(count); i++ {
+		raw := &rawInfos[i]
+
+		// Convert temperature sensors, filtering out unused ones
+		sensors := make([]float64, 0, 8)
+		for _, temp := range raw.temp_sensors {
+			if temp != 0 {
+				sensors = append(sensors, float64(temp))
+			}
+		}
+
+		// todo likely not entirely correct
```

Comment:
?

Reply:
It's mostly correct at least, if it's not we'll probably learn later on